pkg/controller/pod_test.go

```diff
@@ -244,6 +245,58 @@ var _ = Describe("Dynamic Attachment controller", func() {
 					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
 				})
 			})
+
+			When("an attachment is added with attributes (IPs, MAC, cni-args)", func() {
+				JustBeforeEach(func() {
+					pod := updatePodSpec(pod)
+					netSelectionElements := append(generateNetworkSelectionElements(namespace, networkName),
+						nad.NetworkSelectionElement{
+							Name:             networkToAdd,
+							Namespace:        namespace,
+							InterfaceRequest: "net1",
+							IPRequest:        []string{ipAddr},
+							MacRequest:       macAddr,
+							CNIArgs: &map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					)
+					serelizedNetSelectionElements, _ := json.Marshal(netSelectionElements)
+					pod.Annotations[nad.NetworkAttachmentAnnot] = string(serelizedNetSelectionElements)
+					_, err := k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						pod,
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("an `AddedInterface` event is seen in the event recorded", func() {
+					expectedEventPayload := fmt.Sprintf(
+						"Normal AddedInterface pod [%s]: added interface %s to network: %s (ips: [172.16.0.1], mac: 02:03:04:05:06:07, cni-args: &map[foo:bar])",
```

Comment:
might as well template this string right ? At least the ip / mac.

For the cni-args I'd use a struct and implement the stringer interface there, but ... Feels Like going overboard.

Reply:
I just changed it. 
For the CI, I updated the golangci-lint version and fix 1 issue.